frontend/components/JoinGraveBtn.tsx

```diff
@@ -638,16 +663,59 @@ export default function JoinGraveBtn({
       );
     } else {
       return (
-        <Button
-          onClick={handleJoin}
-          disabled={loading}
-          mb="10px"
-          fontFamily="Bungee"
-          fontSize="16px"
-          fontWeight="400"
-        >
-          {loading ? 'Processing...' : 'START'}
-        </Button>
+        <>
+          <AlertDialog
```

Comment:
let's maybe move this to separate component. this file is starting to get quite large